File: rb/features.rb
# Kmail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module KmailFeatures
  def self.make_feature(name)
    case name
    when "base"
      KmailBaseFeature.new
    when "test"
      KmailTestFeature.new
    else
      KmailBaseFeature.new
    end
  end
end

File: py/features.py
# Kmail SDK feature factory

from feature.base_feature import KmailBaseFeature
from feature.test_feature import KmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KmailBaseFeature(),
        "test": lambda: KmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
